shell.nix

```diff
@@ -0,0 +1,105 @@
+{ sources ? import ./nix/sources.nix }:
```

Comment:
I don't know anything about nix, so I can't really review this part.  But I'm wondering if this file can also live in the nix/ directory -- I'm guessing not.

Reply:
This one file is most useful if it's in the root of the repository.  You activate it by running `nix-shell`, which looks for shell.nix in your current directory.  It's also not strictly necessary to include any of the nix bits in this pull request, but I found it very helpful to use nix for pinning all the external dependencies.

Reply:
OK, that's fine then.

Which reminds me: this PR should also update the README to give instructions on how to run the server manually.  It looks like you'll need to install nix on your machine first?  And then run what?

And also instructions on how to update the nix configs if that's something that might ever be needed.  Like, if we add a feature that requires a new OS package to be installed, what do we do to get that to work in nix-land?  Ideally something that's understandable even by people who don't know anything about nix.